### ISSUE TITLE
API Reference Documentation

### ISSUE THREAD

**Issue by andredumas** [andredumas/techan.js#1]
Comment and vote here if you want documentation, otherwise I'll just be focusing on adding new features and enhancements.

**Comment by prionkor** [comment]
I will help on adding new doc. Just started with the lib. Need some more time to get familiarized with it.

**Comment by jreim** [comment]
This is awesome stuff, i will help with documentation. I am also working on custom kockout js bindings, to make the charts dynamic. Maybe i can fork this project and share that once it is completed?

**Comment by andredumas** [comment]
That would be great! Re custom bindings and other extensions, I want to keep techanjs pure, light and simple with minimal dependencies (only d3). Other than core D3 patterns I didn't want to impose particular style or integration onto the user. 

To support custom extensions I was thinking of creating separate adapter projects for this. Eg, techan-angular, techan-knockout, techan-ext-indicators, techan-indicator-seed, techan-angular-seed, etc. The only change I would make in the core is to ensure it plays nicely with other adapters, but I would not want the core design to be dictated by this. Everything can be brought together via bower or npm. 

Feel free to create other issues as discussion points for all to contribute. Maybe kick one off... techan-knockout?

**Comment by jreim** [comment]
yes i totally agree that it should be forked and keep the core in tact. it
is a REALLY nice library. I just started working on this a few days ago, so
i will keep you posted as to my progress and as soon as i get something
semi stable i will throw it out on git for others to play with and check
out.

thanks!
Jesse

**Comment by jreim** [comment]
Hey Andre,

I had a quick question about your library. I am trying to plot my own algos
and indicators (Bollinger bands) for example. But i can quite seem to
figure out how to get them integrated. I have started to add the studies in
the techan.js file (the big unified one) but keep running into an issue
when trying to set up the accessor, specifically setting up the accessor.d
value for the x plot.

Basically what i am doing is the algo calculations server side and then
pulling them in (with a dime field that corresponds to the study). and then
plot the line. Maybe this is already there and i am just not properly using
it.

i added this inside the plot

upperBand: line(accessor.value, plot, plotMixin)

and then i set teh accessor in my code, but i then it barfs at
 svgLine.init(p.accessor.d, p.xScale, p.accessor, p.yScale); b/c
p.accessor.d is undefined.

anyways, i know this is a side thing for you and i really appreciate all of
the work and any help, i think this is a really great lib and hope
to contribute to it with teh knock stuff i am making at some point (when
its less hacked up).

thanks!
Jesse
jessereim@gmail.com


On Wed, Apr 1, 2015 at 8:50 AM, jesse reim <jessereim@gmail.com> wrote:
>
> yes i totally agree that it should be forked and keep the core in tact.
it is a REALLY nice library. I just started working on this a few days ago,
so i will keep you posted as to my progress and as soon as i get something
semi stable i will throw it out on git for others to play with and check
out.
>
> thanks!
> Jesse
>
> On Wed, Apr 1, 2015 at 6:37 AM, André Dumas <notifications@github.com>
wrote:
>>
>> That would be great! Re custom bindings and other extensions, I want to
keep techanjs pure, light and simple with minimal dependencies (only d3).
Other than core D3 patterns I didn't want to impose particular style or
integration onto the user.
>>
>> To support custom extensions I was thinking of creating separate adapter
projects for this. Eg, techan-angular, techan-knockout,
techan-ext-indicators, techan-indicator-seed, techan-angular-seed, etc. The
only change I would make in the core is to ensure it plays nicely with
other adapters, but I would not want the core design to be dictated by
this. Everything can be brought together via bower or npm.
>>
>> Feel free to create other issues as discussion points for all to
contribute. Maybe kick one off... techan-knockout?

**Comment by andredumas** [comment]
Hi @jreim 

Prob best if you open up a specific issue for your question so it doesn't get lost in this one.

My advice if you're creating your own custom plots is to not rely on techanjs' internal APIs. They could change at any time. If you already have data for Bollinger Bands, plotting the 3 lines would be really simple d3. You also don't need the added complexity of accessors as you already know the format of your data. Accessors are there so that techanjs plots can work with any backing data structure.

In short, I wouldn't 'integrate' your plots with the core, use the core to setup some basics (financetime, candlestick) then use standard d3 for your plots.

cheers
andre

**Comment by Kurtas** [comment]
The API documentation is necessary for any open source project. If there isn't api documentation  then it will be always a "one man project". I really want to participate in development, but it's hard without documentation.

**Comment by thomashan** [comment]
@Kurtas 
Maybe you can help out on the documentation :+1:

**Comment by Chidan** [comment]
+1, yes please documentation is very much required and appreciated..

**Comment by yuezhizizhang** [comment]
TechanJS is so amazing! Yes, we need Bollinger Band
